pkg/compact/retention.go

```diff
@@ -0,0 +1,63 @@
+package compact
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/block"
+	"github.com/improbable-eng/thanos/pkg/objstore"
+	"github.com/pkg/errors"
+)
+
+// RetentionPolicy describes retention policy.
+type RetentionPolicy interface {
+	Apply(context.Context) error
+}
+
+// TimeBasedRetentionPolicy is a retention policy based on blocks age.
+type TimeBasedRetentionPolicy struct {
+	logger            log.Logger
+	bkt               objstore.Bucket
+	retentionDuration time.Duration
+}
+
+// NewTimeBasedRetentionPolicy creates new TimeBasedRetentionPolicy.
+func NewTimeBasedRetentionPolicy(logger log.Logger, bkt objstore.Bucket, retentionDuration time.Duration) *TimeBasedRetentionPolicy {
+	return &TimeBasedRetentionPolicy{
+		logger:            logger,
+		bkt:               bkt,
+		retentionDuration: retentionDuration,
+	}
+}
+
+// Apply removes blocks older than rententionDuration based on blocks MaxTime.
+func (tbr *TimeBasedRetentionPolicy) Apply(ctx context.Context) error {
```

Comment:
We are adding interface and whole struct to support just one method.

Can we keep it simple, so just function?
1. We are adding interface even though we don't need it (no logic that requires it)
2. We have no idea what other retention could be there apart time based?
3. Just single function is sufficent here

Any thoughts?